src/main/scala/org/broadinstitute/dsde/rawls/jobexec/SubmissionMonitorActor.scala

```diff
@@ -0,0 +1,283 @@
+                             val submissionId: UUID,
+                             val datasource: SlickDataSource,
+                             val executionServiceDAO: ExecutionServiceDAO,
+                             val credential: Credential,
+                             val submissionPollInterval: FiniteDuration) extends Actor with SubmissionMonitor with LazyLogging {
+
+  import context._
+
+  scheduleNextMonitorPass
+
+  override def receive = {
+    case StartMonitorPass =>
+      logger.debug(s"polling workflows for submission $submissionId")
+      queryExecutionServiceForStatus() pipeTo self
+    case response: ExecutionServiceStatusResponse =>
+      logger.debug(s"handling execution service response for submission $submissionId")
+      handleStatusResponses(response) pipeTo self
+    case StatusCheckComplete(terminateActor) =>
+      logger.debug(s"done checking status for submission $submissionId, terminateActor = $terminateActor")
+      if (terminateActor) stop(self)
+      else scheduleNextMonitorPass
+
+    case Status.Failure(t) => throw t // an error happened in some future, let the supervisor handle it
+  }
+
+  def scheduleNextMonitorPass: Cancellable = {
+    system.scheduler.scheduleOnce(submissionPollInterval, self, StartMonitorPass)
+  }
+
+}
+
+trait SubmissionMonitor extends FutureSupport with LazyLogging {
+  val workspaceName: WorkspaceName
+  val submissionId: UUID
+  val datasource: SlickDataSource
+  val executionServiceDAO: ExecutionServiceDAO
+  val credential: Credential
+  val submissionPollInterval: Duration
+
+  import datasource.dataAccess.driver.api._
+
+  /**
+   * This function starts a monitoring pass
+   * @param executionContext
+   * @return
+   */
+  def queryExecutionServiceForStatus()(implicit executionContext: ExecutionContext): Future[ExecutionServiceStatusResponse] = {
+    datasource.inTransaction { dataAccess =>
+      dataAccess.workflowQuery.listWorkflowRecsForSubmissionAndStatuses(submissionId, WorkflowStatuses.runningStatuses: _*)
+    } flatMap { externalWorkflowIds =>
+      Future.traverse(externalWorkflowIds) { workflowRec =>
+        // for each workflow query the exec service for status and if has Succeeded query again for outputs
+        toFutureTry(execServiceStatus(workflowRec) flatMap execServiceOutputs)
+      }
+    } map (ExecutionServiceStatusResponse)
+  }
+
+  private def execServiceStatus(workflowRec: WorkflowRecord)(implicit executionContext: ExecutionContext): Future[Option[WorkflowRecord]] = {
+    executionServiceDAO.status(workflowRec.externalId, getUserInfo).map(newStatus => {
+      if (newStatus.status != workflowRec.status) Option(workflowRec.copy(status = newStatus.status))
+      else None
+    })
+  }
+
+  private def execServiceOutputs(workflowRecOption: Option[WorkflowRecord])(implicit executionContext: ExecutionContext): Future[Option[(WorkflowRecord, Option[ExecutionServiceOutputs])]] = {
+    workflowRecOption match {
```

Comment:
Can this match be moved up to queryExecutionServiceForStatus?  It would be a little clearer if this method took a WorkflowRecord without the Option.